rotkehlchen/chain/ethereum/modules/uniswap/v3/utils.py

```diff
@@ -0,0 +1,463 @@
+from rotkehlchen.chain.ethereum.contracts import EthereumContract
+from rotkehlchen.chain.ethereum.interfaces.ammswap.types import LiquidityPoolAsset
+from rotkehlchen.chain.ethereum.interfaces.ammswap.utils import TokenDetails
+from rotkehlchen.chain.ethereum.modules.uniswap.v3.types import NFTLiquidityPool
+from rotkehlchen.chain.ethereum.utils import multicall_2
+from rotkehlchen.constants.ethereum import (
+    UNISWAP_V3_FACTORY,
+    UNISWAP_V3_NFT_MANAGER,
+    UNISWAP_V3_POOL_ABI,
+)
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.misc import NotERC20Conformant, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEthAddress
+from rotkehlchen.utils.misc import get_chunks
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+UNISWAP_V3_POSITIONS_PER_CHUNK = 45
+POOL_INIT_CODE_HASH = '0xe34f199b19b2b4f47f68442619d555527d244f78a3297ea89325f843f87b8b54'
+UNISWAP_V3_ERROR_MSG = 'Remote error calling multicall contract for uniswap v3 {} for address properties: {}'  # noqa: 501
+POW_96 = 2**96
+LOG_PRICE = FVal('1.0001')
+
+
+def uniswap_v3_lp_token_balances(
+    userdb: 'DBHandler',
+    address: ChecksumEthAddress,
+    ethereum: 'EthereumManager',
+    price_known_assets: Set[EthereumToken],
+    price_unknown_assets: Set[EthereumToken],
+) -> List[NFTLiquidityPool]:
+    """
+    Fetches all the Uniswap V3 LP positions for the specified address.
+    1. Use the NFT Positions contract to call the `balanceOf` method to get number of positions.
+    2. Loop through from 0 to (positions - 1) using the index and address to call
+    `tokenOfOwnerByIndex` method which gives the NFT ID that represents a LP position.
+    3. Use the token ID gotten above to call the `positions` method to get the current state of the
+    liquidity position.
+    4. Use the `positions` data to generate the LP address using the `compute_pool_address` method.
+    5. Use the pool contract of the addresses generate and call the `slot0` method to get the
+    LP state.
+    6. Get basic information of the tokens in the LP pairs.
+    7. Calculate the price ranges for which the LP position is valid for.
+    8. Calculate the amount of tokens in the LP position.
+    9. Calculate the amount of tokens in the LP.
+
+    If the multicall fails due to `RemoteError` or one of the calls isn't successful, it is omitted
+    from the chunk.
+    """
+    nft_manager_contract = EthereumContract(
+        address=UNISWAP_V3_NFT_MANAGER.address,
+        abi=UNISWAP_V3_NFT_MANAGER.abi,
+        deployed_block=UNISWAP_V3_NFT_MANAGER.deployed_block,
+    )
+    balances: List[NFTLiquidityPool] = []
+    try:
+        my_positions = nft_manager_contract.call(
```

Comment:
Calling this `amount_of_positions` would make it clearer for a new reader to know what this contract call retruns
```suggestion
        amount_of_positions = nft_manager_contract.call(
```

Reply:
okay